packages/store/src/lib/test/AtomMap.test.ts

```diff
@@ -0,0 +1,563 @@
+
+describe('AtomMap', () => {
+	let cleanupFns: (() => void)[] = []
+	afterEach(() => {
+		for (const fn of cleanupFns) {
+			fn()
+		}
+		cleanupFns = []
+	})
+
+	function testReactor(name: string, fn: () => any) {
+		const cb = jest.fn(fn)
+		const cleanup = react(name, cb)
+		cleanupFns.push(() => cleanup())
+		return cb
+	}
+
+	describe('get', () => {
+		it('should return the value or undefined', () => {
+			const map = new AtomMap('test')
+			expect(map.get('a')).toBe(undefined)
+
+			map.set('a', 1)
+			expect(map.get('a')).toBe(1)
+
+			map.delete('a')
+			expect(map.get('a')).toBe(undefined)
+		})
+
+		it('should react to creating, updating, and deleting values', () => {
+			const map = new AtomMap('test', [['a', 1]])
+			const reactor = testReactor('test', () => map.get('b'))
+			expect(reactor).toHaveBeenCalledTimes(1)
+			expect(reactor).toHaveLastReturnedWith(undefined)
+
+			map.set('b', 2)
+			expect(reactor).toHaveBeenCalledTimes(2)
+			expect(reactor).toHaveLastReturnedWith(2)
+
+			map.set('b', 3)
+			expect(reactor).toHaveBeenCalledTimes(3)
+			expect(reactor).toHaveLastReturnedWith(3)
+
+			map.delete('b')
+			expect(reactor).toHaveBeenCalledTimes(4)
+			expect(reactor).toHaveLastReturnedWith(undefined)
+		})
+
+		it('should not react to changes in other keys', () => {
+			const map = new AtomMap('test', [['a', 1]])
+			const aReactor = testReactor('a-reactor', () => map.get('a'))
+			expect(aReactor).toHaveBeenCalledTimes(1)
+
+			// Setting other keys shouldn't trigger a reaction
+			map.set('b', 1)
+			map.set('c', 2)
+			expect(aReactor).toHaveBeenCalledTimes(1)
+
+			// Updating other keys shouldn't trigger a reaction
+			map.set('b', 3)
+			map.set('c', 4)
+			expect(aReactor).toHaveBeenCalledTimes(1)
+
+			// Deleting other keys shouldn't trigger a reaction
+			map.delete('b')
+			map.delete('c')
+			expect(aReactor).toHaveBeenCalledTimes(1)
+
+			// But setting the watched key should trigger a reaction
+			map.set('a', 5)
+			expect(aReactor).toHaveBeenCalledTimes(2)
+		})
+	})
+
+	describe('has', () => {
+		it('should return true if the key exists with a value', () => {
+			const map = new AtomMap('test')
+			expect(map.has('a')).toBe(false)
+
+			map.set('a', 1)
+			expect(map.has('a')).toBe(true)
+
+			map.delete('a')
+			expect(map.has('a')).toBe(false)
+		})
+
+		it('should react to creating, updating, and deleting values', () => {
+			const map = new AtomMap('test')
+			const reactor = testReactor('test', () => map.has('a'))
+			expect(reactor).toHaveBeenCalledTimes(1)
+			expect(reactor).toHaveLastReturnedWith(false)
+
+			map.set('a', 1)
+			expect(reactor).toHaveBeenCalledTimes(2)
+			expect(reactor).toHaveLastReturnedWith(true)
+
+			map.set('a', 2)
+			expect(reactor).toHaveBeenCalledTimes(3) // setting a value triggers a reaction
+			expect(reactor).toHaveLastReturnedWith(true)
+
+			map.delete('a')
+			expect(reactor).toHaveBeenCalledTimes(4)
+			expect(reactor).toHaveLastReturnedWith(false)
+		})
```

Comment:
i'd be interested in checking that the reactivity remains after deletion to make sure reinsertions work
```suggestion

			map.set('a', 2)
			expect(reactor).toHaveBeenCalledTimes(5) // reinstating a previously deleted atom works
			expect(reactor).toHaveLastReturnedWith(true)
		})
```